staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go

```diff
@@ -18,30 +18,22 @@ package routes
 
 import (
 	"github.com/emicklei/go-restful"
+	"github.com/golang/glog"
 
-	"k8s.io/apimachinery/pkg/openapi"
 	"k8s.io/apiserver/pkg/server/mux"
-	apiserveropenapi "k8s.io/apiserver/pkg/server/openapi"
-
-	"github.com/golang/glog"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/handler"
 )
 
 // OpenAPI installs spec endpoints for each web service.
 type OpenAPI struct {
-	Config *openapi.Config
+	Config *common.Config
 }
 
 // Install adds the SwaggerUI webservice to the given mux.
-func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) *apiserveropenapi.OpenAPIService {
-	openapiSpec, err := apiserveropenapi.BuildSwaggerSpec(c.RegisteredWebServices(), oa.Config)
-	if err != nil {
-		glog.Fatalf("Failed to register open api spec for root: %v", err)
-		return nil
-	}
-	service, err := apiserveropenapi.RegisterOpenAPIService(openapiSpec, "/swagger.json", mux)
+func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) {
+	_, err := handler.BuildAndRegisterOpenAPIService("/swagger.json", c.RegisteredWebServices(), oa.Config, mux)
```

Comment:
`c` is only for one of the servers in a delegation chain. What about the others? They also have webservices we have to consider (btw, the same for swagger).